lib/exqlite/connection.ex

```diff
@@ -652,12 +652,12 @@ defmodule Exqlite.Connection do
 
   defp bind_params(%Query{ref: ref, statement: statement} = query, params, state)
        when ref != nil do
-    case Sqlite3.bind(state.db, ref, params) do
-      :ok ->
-        {:ok, query}
-
-      {:error, reason} ->
-        {:error, %Error{message: to_string(reason), statement: statement}, state}
+    try do
+      Sqlite3.bind(ref, params)
+    rescue
+      e -> {:error, %Error{message: Exception.message(e), statement: statement}, state}
```

Comment:
We lose useful stacktrace here (which `bind_*` was called? helps identify which type was expected and what value was received), but this can probably be handled later.

Reply:
Can we reraise with a reformed error and retain the stack trace?

Reply:
Maybe. I don't have much context on this function or how DBConnection handles exceptions from within .... I think it's safer for now to keep the previous behavior of not raising and returning an error triplet. I tried to think of various scenarios in https://github.com/elixir-sqlite/exqlite/pull/298#discussion_r1792817319 and right now I don't see anything useful being lost.